listenbrainz/webserver/static/js/src/listens/ListenCard.tsx

```diff
@@ -0,0 +1,207 @@
+import * as timeago from "time-ago";
+
+import * as React from "react";
+import { get as _get } from "lodash";
+import MediaQuery from "react-responsive";
+import {
+  faMusic,
+  faHeart,
+  faHeartBroken,
+} from "@fortawesome/free-solid-svg-icons";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import { getArtistLink, getTrackLink, searchForSpotifyTrack } from "../utils";
+import Card from "../components/Card";
+import APIService from "../APIService";
+
+const DEFAULT_COVER_ART_URL = "/static/img/default_cover_art.png";
+
+type ListenCardProps = {
+  apiUrl: string;
+  listen: Listen;
+  mode: ListensListMode;
+  className?: string;
+  playListen: (listen: Listen) => void;
+};
+
+type ListenCardState = {
+  coverArtUrl?: string;
+  feedback: ListenFeedBack;
+};
+
+export default class ListenCard extends React.Component<
+  ListenCardProps,
+  ListenCardState
+> {
+  private APIService: APIService;
+  playListen: (listen: Listen) => void;
+
+  constructor(props: ListenCardProps) {
+    super(props);
+
+    this.playListen = props.playListen.bind(this, props.listen);
```

Comment:
If we use arrow functions we don't have to worry about binding them in constructor.